lnwallet/channel_test.go

```diff
@@ -8025,60 +8031,165 @@ func TestChannelMaxFeeRate(t *testing.T) {
 	// propertyTest tests that the validateFeeRate function always passes
 	// for the output returned by MaxFeeRate for any valid random inputs
 	// fed to MaxFeeRate.
-	propertyTest := func(c *LightningChannel) func(alloc maxAlloc,
-		am anchorMax) bool {
+	propertyTest := func(ma maxAlloc) bool {
+		maxFeeRate := aliceChannel.MaxFeeRate(float64(ma))
+		return aliceChannel.validateFeeRate(maxFeeRate) == nil
+	}
 
-		return func(ma maxAlloc, am anchorMax) bool {
-			maxFeeRate := c.MaxFeeRate(
-				float64(ma), chainfee.SatPerKWeight(am),
-			)
+	if err = quick.Check(propertyTest, nil); err != nil {
+		t.Fatal(err)
+	}
 
-			return c.validateFeeRate(maxFeeRate) == nil
+	assertMaxFeeRate(aliceChannel, 1.0, 676794154)
+	assertMaxFeeRate(aliceChannel, 0.001, 676794)
+	assertMaxFeeRate(aliceChannel, 0.000001, 676)
+	assertMaxFeeRate(aliceChannel, 0.0000001, chainfee.FeePerKwFloor)
+}
+
+// TestIdealCommitFeeRate tests that we correctly compute the ideal commitment
+// fee of a channel given the current network fee, minimum relay fee, maximum
+// fee allocation and whether or not the channel has anchor outputs.
+func TestIdealCommitFeeRate(t *testing.T) {
+	t.Parallel()
+
+	assertIdealFeeRate := func(c *LightningChannel, netFee, minRelay,
+		maxAnchorCommit chainfee.SatPerKWeight,
+		maxFeeAlloc float64, expectedFeeRate chainfee.SatPerKWeight) {
+
+		feeRate := c.IdealCommitFeeRate(
+			netFee, minRelay, maxAnchorCommit, maxFeeAlloc,
+		)
+		if feeRate != expectedFeeRate {
+			t.Fatalf("expected fee rate of %v got %v",
+				expectedFeeRate, feeRate)
 		}
-	}
 
-	aliceChannel, _, cleanUp, err := CreateTestChannels(
-		channeldb.SingleFunderTweaklessBit,
-	)
-	if err != nil {
-		t.Fatalf("unable to create test channels: %v", err)
+		if err := c.validateFeeRate(feeRate); err != nil {
+			t.Fatalf("fee rate validation failed: %v", err)
+		}
 	}
-	defer cleanUp()
 
-	if err = quick.Check(propertyTest(aliceChannel), nil); err != nil {
-		t.Fatal(err)
+	// propertyTest tests that the validateFeeRate function always passes
+	// for the output returned by IdealCommitFeeRate for any valid random
+	// inputs fed to MaxFeeRate.
+	propertyTest := func(c *LightningChannel) func(ma maxAlloc) bool {
+		return func(ma maxAlloc) bool {
+			maxFeeRate := c.MaxFeeRate(float64(ma))
```

Comment:
Should this use `IdealCommitFeeRate` instead? O/w it's the same as the other property test

Reply:
Indeed! good catch